core/server/master/src/main/java/alluxio/master/throttle/MetricsMonitorUtils.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.throttle;
+
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import java.lang.management.BufferPoolMXBean;
+
+/**
+ * The Metrics monitor utils.
+ */
+public class MetricsMonitorUtils {
+  private static BufferPoolMXBean sBufferPool;
+
+  static {
+    for (BufferPoolMXBean bufferPoolMXBean
+        : sun.management.ManagementFactoryHelper.getBufferPoolMXBeans()) {
+      if (bufferPoolMXBean.getName().equals("direct")) {
+        sBufferPool = bufferPoolMXBean;
+      }
```

Comment:
I wonder if this should be moved to OperationSystemGaugeSet.java and created as a metric there?

Reply:
Not exact, the OS serial reflects the OS status. While this mem is just about the direct allocated buffer.

Reply:
I guess I'm wondering why this isn't a metric like all the other ones. Would it not be useful to add this as a metric in the metrics package? Or do you think it only belongs here?

Reply:
let me check today

Reply:
So far it is not exported, I am worrying about the performance impact, I put it there. Do you think it is necessary to move it to the metrics system?

Reply:
will update in the following pr